backend/src/test/java/com/example/backend/controller/PlayerControllerTest.java

```diff
@@ -0,0 +1,102 @@
+package com.example.backend.controller;
+
+import com.example.backend.model.Player;
+import com.example.backend.repository.PlayerRepo;
+import com.example.backend.service.IdService;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class PlayerControllerTest {
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @Autowired
+    private PlayerRepo repo;
+
+    @MockBean
+    private IdService idService;
+
+    @DirtiesContext
+    @Test
+    void getAllPlayers() throws Exception {
+        //Given
+        Player dummyPlayer = new Player("1337","Spieler1");
+        repo.save(dummyPlayer);
+
+        //When & Then
+        mockMvc.perform(
+                get("/api/players"))
+                .andExpect(status().is(200))
+                .andExpect(content().string("""
+                        [{"id":"1337","playerName":"Spieler1"}]"""));
+    }
+
+    @DirtiesContext
+    @Test
+    void getPlayer_whenPlayerNotExists_returns404() throws Exception {
+        //Given
+
+        //When & Then
+        mockMvc.perform(
+                get("/api/players/1337/abc"))
+                .andExpect(status().is(404));
+    }
+
+    @Test
+    void addPlayer() throws  Exception{
```

Comment:
Consider adding a more specific Method name. What is the Test supposed to return?